<a href="https://colab.research.google.com/github/ElofssonLab/kb8029-book/blob/main/notebooks/session05-discussion-3.ipynb" style="display:inline-block;padding:10px 18px;background-color:#F9AB00;color:#000000;font-weight:bold;text-decoration:none;border-radius:6px;font-family:sans-serif;font-size:14px;">&#9654;&nbsp; Open in Google Colab</a>

# Session 5 — In-class discussion problem (3 of 3)

Work this out **by hand in your group first** — then run the code cell to check your answer before presenting.

## One protein, three domains, three very different E-values

The book page's Pfam example used *HBB* and found one domain (Globin, E-value ~1.3&times;10&#8315;&sup2;&#8319;). **BRCA1** (UniProt `P38398`, the breast-cancer susceptibility gene) is a much larger protein with **three** separate Pfam domain hits, with very different E-values.

1. Without running anything: what could make one domain's match far more significant (a much smaller E-value) than another's on the same protein? Think about the length of the domain, and about how similar the members of each Pfam family are to each other.
2. Predict: does a less extreme (but still small) E-value mean the domain is less real, or that the evidence from sequence alone is weaker?
3. Run the cell below, then rank the three matches from most to least significant. Compare each E-value with the length of the matched region.

In [1]:
import requests

r = requests.get("https://www.ebi.ac.uk/interpro/api/entry/pfam/protein/uniprot/P38398/", timeout=10)
r.raise_for_status()
results = r.json()["results"]

print(f"{'Pfam ID':<10}{'Name':<45}{'Region':<15}{'Length':>8}{'E-value':>12}")
for entry in results:
    meta = entry["metadata"]
    loc = entry["proteins"][0]["entry_protein_locations"][0]   # "score" here is the domain E-value
    frag = loc["fragments"][0]
    region = f"{frag['start']}-{frag['end']}"
    print(f"{meta['accession']:<10}{meta['name']:<45}{region:<15}{frag['end'] - frag['start'] + 1:>8}{loc['score']:>12}")

Pfam ID   Name                                         Region           Length     E-value
PF00097   Zinc finger, C3HC4 type (RING finger)        24-64                41     1.6e-06
PF00533   BRCA1 C Terminus (BRCT) domain               1645-1723            79     4.6e-07
PF12820   Serine-rich domain associated with BRCT      345-508             164     2.8e-77


**Discussion point:** an E-value measures how surprising the match is, not how "real" or important the domain is. BRCA1 genuinely has a RING finger, BRCT domains, and the serine-rich region next to them. The serine-rich domain (PF12820, about 160 residues) matches with an E-value of ~2.8&times;10&#8315;&sup7;&sup7;: it is long, and the family consists of BRCA1 orthologs that are closely related, so the profile is very specific. The RING finger (PF00097, about 40 residues) gets a far more modest ~1.6&times;10&#8315;&#8310;: the domain is short, so there are few positions to add up score over, and the RING family is very diverse — apart from the zinc-binding cysteines and histidines, its members share little sequence. Match significance depends on the length of the domain, how conserved and diverse the family is, and how well this sequence aligns to the model. A weaker but still significant E-value is not "probably not there"; it is weaker evidence from sequence alone, which other evidence (a structure, the conserved zinc-binding residues) can strengthen.